freedomofpress/securedrop
# Invalidate Session When Admin Resets Journalist Password

## Status

Ready for review

## Description of Changes

Fixes Issue #2300 - Failure to invalidate session when user's password is reset by admin.

Changes proposed in this pull request:

## Testing

This PR adds a new field to the Journalist model to include a "session nonce". This value is authenticated against the database when actions are made, and if it does not match the most recent value, the user is logged out.

I am also not 100% if I correctly wrote my database migration code, so I would feel more comfortable if someone made sure that was correct before pulling this in.

To test this:

1. Log in as a journalist
2. In a seperate browser, login as an admin. Change the password for the journalist.
3. Attempt to perform an action as the journalist in the first browser. It should flash that you have been logged out due to password change.

## Deployment

This commit modifies the models, and alembic migration code is written.

## Checklist

### If you made changes to the server application code:


### If you made non-trivial code changes:
